DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -1,369 +1,424 @@
-                }
-            }
-
-            // Call base method to delete file
-            base.UnInstallFile(scriptFile);
-        }
-
-        private bool ExecuteSql(InstallFile scriptFile)
-        {
-            bool bSuccess = true;
-
-            this.Log.AddInfo(string.Format(Util.SQL_BeginFile, scriptFile.Name));
-
-            // read script file for installation
-            string strScript = FileSystemUtils.ReadFile(this.PhysicalBasePath + scriptFile.FullName);
-
-            // This check needs to be included because the unicode Byte Order mark results in an extra character at the start of the file
-            // The extra character - '?' - causes an error with the database.
-            if (strScript.StartsWith("?"))
-            {
-                strScript = strScript.Substring(1);
-            }
-
-            string strSQLExceptions = DataProvider.Instance().ExecuteScript(strScript);
-            if (!string.IsNullOrEmpty(strSQLExceptions))
-            {
-                if (this.Package.InstallerInfo.IsLegacyMode)
-                {
-                    this.Log.AddWarning(string.Format(Util.SQL_Exceptions, Environment.NewLine, strSQLExceptions));
-                }
-                else
-                {
-                    this.Log.AddFailure(string.Format(Util.SQL_Exceptions, Environment.NewLine, strSQLExceptions));
-                    bSuccess = false;
-                }
-            }
-
-            this.Log.AddInfo(string.Format(Util.SQL_EndFile, scriptFile.Name));
-            return bSuccess;
-        }
-
-        private bool IsValidScript(string fileExtension)
-        {
-            return this.ProviderConfiguration.DefaultProvider.Equals(fileExtension, StringComparison.InvariantCultureIgnoreCase) || fileExtension.Equals("sql", StringComparison.InvariantCultureIgnoreCase);
-        }
-
-        private bool InstallScriptFile(InstallFile scriptFile)
-        {
-            // Call base InstallFile method to copy file
-            bool bSuccess = this.InstallFile(scriptFile);
-
-            // Process the file if it is an Install Script
-            var extension = Path.GetExtension(scriptFile.Name.ToLowerInvariant());
-            if (extension != null)
-            {
-                string fileExtension = extension.Substring(1);
-                if (bSuccess && this.IsValidScript(fileExtension))
-                {
-                    this.Log.AddInfo(Util.SQL_Executing + scriptFile.Name);
-                    bSuccess = this.ExecuteSql(scriptFile);
-                }
-            }
-
-            return bSuccess;
-        }
-    }
-}
+        protected override void UnInstallFile(InstallFile scriptFile)
+        {
+            // Process the file if it is an UnInstall Script
+            if (this.UnInstallScripts.ContainsValue(scriptFile))
+            {
+                if (scriptFile.Name.StartsWith("uninstall.", StringComparison.InvariantCultureIgnoreCase))

```

Comment:
Do we still want this check?  Since we're already checking if it's in `UnInstallScripts` above, we shouldn't need to validate, and we've started supporting `type="uninstall"` without the naming convention (right?)

Reply:
My guess is that it is still there for legacy's sake. If someone didn't label the file with a type uninstall it would still get through. But I'm all in favor of deprecating these quirks and focusing on the types instead.

Reply:
We can deprecate the old behavior in a major version IMO